From 3a8657e43011a24112155a515485f79b5df14aa0 Mon Sep 17 00:00:00 2001
From: kbillore <kaushal.billore@intel.com>
Date: Tue, 21 Jul 2020 20:02:13 +0530
Subject: [PATCH] release usb receiver on destroy

during stability observed crash because of receiver
is not unregistered.

Solution: releasing receiver on application destroy

Tracked-On: OAM-91658
Signed-off-by: Shiva Kumara shiva.kumara.rudrappa@intel.com
Signed-off-by: kbillore <kaushal.billore@intel.com>
---
 .../intel/multicamera/FullScreenActivity.java | 17 ++++++++++++
 .../intel/multicamera/MultiViewActivity.java  | 26 +++++++++++++++++++
 .../multicamera/SingleCameraActivity.java     | 11 +++++++-
 3 files changed, 53 insertions(+), 1 deletion(-)

diff --git a/camera/MultiCameraApplication/java/com/intel/multicamera/FullScreenActivity.java b/camera/MultiCameraApplication/java/com/intel/multicamera/FullScreenActivity.java
index 9e4f731..ed735d5 100644
--- a/camera/MultiCameraApplication/java/com/intel/multicamera/FullScreenActivity.java
+++ b/camera/MultiCameraApplication/java/com/intel/multicamera/FullScreenActivity.java
@@ -301,6 +301,7 @@ public class FullScreenActivity extends AppCompatActivity {
     }
 
     public void openBackCamera() {
+	System.out.println(TAG+"open back camera start");
         closeCamera();
 
         GetCameraCnt();
@@ -317,15 +318,21 @@ public class FullScreenActivity extends AppCompatActivity {
         updateStorageSpace(null);
 
         OpenOnlyBackCamera();
+	System.out.println(TAG+"open back camera end");
+
     }
 
     public void openFrontCamera() {
+	System.out.println(TAG+"open front camera start");
+
         closeCamera();
 
         GetCameraCnt();
         updateStorageSpace(null);
 
         OpenOnlyFrontCamera();
+	System.out.println(TAG+"open front camera end");
+
     }
 
     private void OpenOnlyFrontCamera() {
@@ -456,6 +463,16 @@ public class FullScreenActivity extends AppCompatActivity {
 
     }
 
+    @Override
+    protected void onDestroy() {
+        try {
+            unregisterReceiver(mCameraInst.getmUsbReceiver());
+        }catch (Exception e) {
+            Log.e(TAG, "fatal during unregister receiver");
+        }
+        super.onDestroy();
+    }
+
     @Override
     protected void onPause() {
         super.onPause();
diff --git a/camera/MultiCameraApplication/java/com/intel/multicamera/MultiViewActivity.java b/camera/MultiCameraApplication/java/com/intel/multicamera/MultiViewActivity.java
index 7b3b68d..00d1abe 100644
--- a/camera/MultiCameraApplication/java/com/intel/multicamera/MultiViewActivity.java
+++ b/camera/MultiCameraApplication/java/com/intel/multicamera/MultiViewActivity.java
@@ -300,6 +300,8 @@ public class MultiViewActivity extends AppCompatActivity {
     }
 
     public void Open_TopLeftCam() {
+        System.out.println(TAG+"open top left camera  start");
+
         String[] Data = new String[5];
         ImageButton[] Buttons = new ImageButton[6];
 
@@ -344,9 +346,13 @@ public class MultiViewActivity extends AppCompatActivity {
 
         ic_camera.setTopLeftCam(new CameraBase(this, mTopLeftCam_textureView, Buttons, mRecordingTimeView,
                                      Data, roundedThumbnailView));
+        System.out.println(TAG+"open top left camera  end");
+
     }
 
     public void Open_TopRightCam() {
+        System.out.println(TAG+"open top right camera  start");
+
         String[] Data = new String[5];
         ImageButton[] Buttons = new ImageButton[6];
         mTopRightCam_textureView = findViewById(R.id.textureview1);
@@ -390,9 +396,13 @@ public class MultiViewActivity extends AppCompatActivity {
 
         ic_camera.setTopRightCam(new CameraBase(this, mTopRightCam_textureView, Buttons,
                                       mRecordingTimeView0, Data, roundedThumbnailView));
+        System.out.println(TAG+"open top right camera  end");
+
     }
 
     public void Open_BotmLeftCam() {
+        System.out.println(TAG+"open bot left camera  start");
+
         String[] Data = new String[5];
         ImageButton[] Buttons = new ImageButton[6];
         mBotmLeftCam_textureView = findViewById(R.id.textureview2);
@@ -437,9 +447,13 @@ public class MultiViewActivity extends AppCompatActivity {
 
         ic_camera.setBotLeftCam(new CameraBase(this, mBotmLeftCam_textureView, Buttons,
                                       mRecordingTimeView1, Data, roundedThumbnailView));
+        System.out.println(TAG+"open top left camera  end");
+
     }
 
     public void Open_BotmRightCam() {
+	System.out.println(TAG+"open bot right camera  start");
+
         String[] Data = new String[5];
         ImageButton[] Buttons = new ImageButton[6];
         mBotmRightCam_textureView = findViewById(R.id.textureview3);
@@ -482,6 +496,8 @@ public class MultiViewActivity extends AppCompatActivity {
 
         ic_camera.setBotRightCam(new CameraBase(this, mBotmRightCam_textureView, Buttons,
                                        mRecordingTimeView2, Data, roundedThumbnailView));
+        System.out.println(TAG+"open bot right camera  end");
+
     }
 
     private void manageTopLeftCam() {
@@ -650,6 +666,16 @@ public class MultiViewActivity extends AppCompatActivity {
         }
     }
 
+    @Override
+    protected void onDestroy() {
+        try {
+            unregisterReceiver(ic_camera.getmUsbReceiver());
+        }catch (Exception e) {
+            Log.e(TAG, "fatal during unregister receiver");
+        }
+        super.onDestroy();
+    }
+
     @Override
     protected void onPause() {
         System.out.println("onPause");
diff --git a/camera/MultiCameraApplication/java/com/intel/multicamera/SingleCameraActivity.java b/camera/MultiCameraApplication/java/com/intel/multicamera/SingleCameraActivity.java
index a8c09a7..cef899b 100644
--- a/camera/MultiCameraApplication/java/com/intel/multicamera/SingleCameraActivity.java
+++ b/camera/MultiCameraApplication/java/com/intel/multicamera/SingleCameraActivity.java
@@ -345,7 +345,16 @@ public class SingleCameraActivity extends AppCompatActivity {
 
     }
 
-
+    @Override
+    protected void onDestroy() {
+        try {
+            unregisterReceiver(mCameraInst.getmUsbReceiver());
+        }catch (Exception e) {
+            Log.e(TAG, "fatal during unregister receiver");
+        }
+        super.onDestroy();
+    }
+    
     @Override
     protected void onPause() {
         super.onPause();
-- 
2.17.1

